sympy/physics/control/tests/test_lti.py

```diff
@@ -577,53 +574,186 @@ def test_Feedback_construction():
+    assert tfm6.args == ([[TF1, TF2, TF3], [tf5, tf6, -tf6]], (2, 3), s)
+
+    tfm7 = TransferFunctionMatrix([[TF1, TF2], [TF3, -tf5], [-tf5, TF2]], (3, 2), s)
+    assert tfm7.shape == (tfm7.outputs, tfm7.inputs) == (3, 2)
+    assert tfm7.args == ([[TF1, TF2], [TF3, -tf5], [-tf5, TF2]], (3, 2), s)
+
+    raises(TypeError, lambda: TransferFunction(Matrix([1, 2, 3]), (1, 1), s))
+    raises(ValueError, lambda: TransferFunctionMatrix([TF1, TF2, tf4], (3, 1), s))
+    raises(ValueError, lambda: TransferFunctionMatrix([[TF1], [TF3, tf5]], (2, 2), s))
+    raises(TypeError, lambda: TransferFunctionMatrix([[TF1, TF2], [TF3, Matrix([1, 2])]], (2, 2), s))
+    raises(ValueError, lambda: TransferFunctionMatrix([[TF1, tf4], [TF3, tf5]], (2, 2), s))
+
+
+def test_TransferFunctionMatrix_functions():
+    tf5 = TransferFunction(a1*s**2 + a2*s - a0, s + a0, s)
+    tf6 = TransferFunction(a0*s**3 + a1*s**2 - a2*s, b0*p**4 + b1*p**3 - b2*s*p, s)
+
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    assert -tfm1 == TransferFunctionMatrix([-TF1, -TF2], (2, 1), s)
+
+    tfm2 = TransferFunctionMatrix([[TF1, -TF2], [TF3, tf5]], (2, 2), s)
+    assert -tfm2 == TransferFunctionMatrix([[-TF1, TF2], [-TF3, -tf5]], (2, 2), s)
+
+    tfm3 = TransferFunctionMatrix([[TF1, TF2, TF3], [tf5, -TF1, -TF3]], (2, 3), s)
+    assert -tfm3 == TransferFunctionMatrix([[-TF1, -TF2, -TF3], [-tf5, TF1, TF3]], (2, 3), s)
+
+    tfm4 = TransferFunctionMatrix([TF1, TF2, TF3], (3, 1), s)
+    assert tfm4.is_proper
+    assert not tfm4.is_strictly_proper
+    assert not tfm4.is_biproper
+
+    tfm5 = TransferFunctionMatrix([[TF1, TF2], [TF3, -tf5], [-tf5, TF2]], (3, 2), s)
+    assert not tfm5.is_proper
+    assert not tfm5.is_strictly_proper
+    assert not tfm5.is_biproper
+
+    tfm6 = TransferFunctionMatrix([[TF1, TF2], [TF3, -TF3]], (2, 2), s)
+    assert tfm6.is_proper
+    assert not tfm6.is_strictly_proper
+    assert not tfm6.is_biproper
+
+    tfm7 = TransferFunctionMatrix([-TF1, -tf6], (2, 1), s)
+    assert not tfm7.is_proper
+    assert not tfm7.is_strictly_proper
+    assert not tfm7.is_biproper
+
+
+def test_TransferFunctionMatrix_addition_and_subtraction():
+    tf = TransferFunction(a0*p, a1*p**2 + a2*p - a0, p)
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    tfm2 = TransferFunctionMatrix([-TF2, -TF1], (2, 1), s)
+    tfm3 = TransferFunctionMatrix([TF1, TF1], (2, 1), s)
+    tfm4 = TransferFunctionMatrix([tf], (1, 1), p)
+    tfm5 = TransferFunctionMatrix([TF2, TF1, TF3], (3, 1), s)
+    tfm6 = TransferFunctionMatrix([-TF1, TF2, -TF3], (3, 1), s)
+    tfm7 = TransferFunctionMatrix([TF2], (1, 1), s)
+
+    # addition
+    assert tfm1 + tfm2 == Parallel(tfm1, tfm2)
+    assert tfm3 + tfm1 == Parallel(tfm3, tfm1)
+    assert tfm2 + (tfm3 + tfm1) == Parallel(tfm2, tfm3, tfm1)
+    assert tfm1 + tfm3*tfm7 == Parallel(tfm1, Series(tfm3, tfm7))
+
+    c = symbols("c", commutative=False)
+    raises(ValueError, lambda: tfm1 + Matrix([1, 2, 3]))
```

Comment:
I think this should work if dimensions match. The Matrix should be converted to a compatible TransferFunctionMatrix if possible.

Reply:
If we are not supporting Matrix() input, then I guess there's no point of converting a `Matrix` into a compatible `TransferFunctionMatrix` here.

Reply:
We are implicitly supportin Matrix because it acts like a lists of lists and can contain TransferFunction objects (currently). No error should be raised for `TransferFunctionMatrix(Matrix([[TF, TF],[TF,TF]]))`.

Reply:
Unfortunately that's not the case for sympy's Matrix which iterates differently to numpy's 2D arrays or the deprecated `np.matrix`:
```julia
In [27]: npa = np.array([[1, 2], [3, 4]])                                                                                                      

In [28]: npm = np.matrix([[1, 2], [3, 4]])                                                                                                     

In [29]: sym = Matrix([[1, 2], [3, 4]])                                                                                                        

In [30]: list(npa)                                                                                                                             
Out[30]: [array([1, 2]), array([3, 4])]

In [31]: list(npm)                                                                                                                             
Out[31]: [matrix([[1, 2]]), matrix([[3, 4]])]

In [32]: list(sym)                                                                                                                             
Out[32]: [1, 2, 3, 4]
```
Both numpy versions iterate over rows with the rows represented as 1D arrays are 1xn matrices. Iterating over a sympy Matrix yields the elements of the flattened matrix analogous to `np.array.flat`.